pytorch_lightning/strategies/ddp_spawn.py

```diff
@@ -241,20 +241,20 @@ def training_step(self, *args, **kwargs) -> STEP_OUTPUT:
 
     def validation_step(self, *args, **kwargs) -> Optional[STEP_OUTPUT]:
         with self.precision_plugin.val_step_context():
-            if isinstance(self.model, DistributedDataParallel):
+            if self.lightning_module.trainer.state.fn == TrainerFn.FITTING:
                 # used when calling `trainer.fit`
                 return self.model(*args, **kwargs)
             else:
                 # used when calling `trainer.validate`
-                return self.lightning_module.validation_step(*args, **kwargs)
+                return self.model.validation_step(*args, **kwargs)
```

Comment:
took me a moment to understand. 
here, model is not wrapped in DDP, but it's potentially wrapped in a double precision module, which is why we should reference self.model here. 

But this will only work if the wrapper implements the x_step methods, so this change here now introduces a typing issue. 

On a larger scope, now that the precision plugin is inside the strategy, I believe the precision plugin should be involved here directly instead of wrapping the model.

Reply:
#13000

Reply:
yep. sounds reasonable.